### ISSUE TITLE
[feature] Add pager for interactive search of multiple line match and context match

### ISSUE THREAD

**Issue by HerringtonDarkholme** [ast-grep/ast-grep#548]
# ⭐ Suggestion

ast-grep currently outputs al matches in a detached screen for interactive mode output.

The screen can help us separate different results across files. However, matches in one file can still be quite overwhelming for one single screen.

Piping the output in a pager will be more readable. 
https://docs.rs/minus/latest/minus/index.html looks like a promising library.

# 💻 Use Cases

Improve readability of printing code around match, multiple line matches, or many matches in one file.

**Comment by kqvanity** [comment]
Would that allow it be similar to igrep?

**Comment by HerringtonDarkholme** [comment]
@kqvanity no, igrep handles changing query. here the issue only addresses a pager functionality like `less` (page up/down)

**Comment by HerringtonDarkholme** [comment]
Nope, the interactive result is left for the ecosystem to implement.  e.g. https://github.com/MagicDuck/grug-far.nvim

**Comment by HerringtonDarkholme** [comment]
This feature needs more user feedback to be considered. Proof of concept implementation is easy but integrating into ast-grep needs quite a lot of work.
